From b0890d371ca1e49877c45d696deb9102f1b605c3 Mon Sep 17 00:00:00 2001
From: Glenn Washburn <development@efficientek.com>
Date: Wed, 26 Jul 2023 16:02:36 -0500
Subject: [PATCH 12/57] term/ns8250-spcr: Continue processing SPCR table even
 if revision is < 2

According to commit 0231d00082 (ACPI: SPCR: Make SPCR available to x86)
to the Linux kernel, "On x86, many systems have a valid SPCR table but the
table version is not 2 so the table version check must be a warning."

Signed-off-by: Glenn Washburn <development@efficientek.com>
Reviewed-by: Benjamin Herrenschmidt <benh@kernel.crashing.org>
Reviewed-by: Daniel Kiper <daniel.kiper@oracle.com>
---
 grub-core/term/ns8250-spcr.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/grub-core/term/ns8250-spcr.c b/grub-core/term/ns8250-spcr.c
index d101bffb5..d52b52c26 100644
--- a/grub-core/term/ns8250-spcr.c
+++ b/grub-core/term/ns8250-spcr.c
@@ -18,6 +18,7 @@
 
 #if !defined(GRUB_MACHINE_IEEE1275) && !defined(GRUB_MACHINE_QEMU)
 
+#include <grub/misc.h>
 #include <grub/serial.h>
 #include <grub/ns8250.h>
 #include <grub/types.h>
@@ -34,7 +35,8 @@ grub_ns8250_spcr_init (void)
   if (spcr == NULL)
     return NULL;
   if (spcr->hdr.revision < 2)
-    return NULL;
+    grub_dprintf ("serial", "SPCR table revision %d < 2, continuing anyway\n",
+		  (int) spcr->hdr.revision);
   if (spcr->intf_type != GRUB_ACPI_SPCR_INTF_TYPE_16550 &&
       spcr->intf_type != GRUB_ACPI_SPCR_INTF_TYPE_16550X)
     return NULL;
-- 
2.42.0

